presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -958,6 +959,16 @@ private GroupedExecutionProperties processWindowFunction(PlanNode node)
             return GroupedExecutionProperties.notCapable();
         }
 
+        @Override
+        public GroupedExecutionProperties visitMarkDistinct(MarkDistinctNode node, Void context)
+        {
+            GroupedExecutionProperties properties = getOnlyElement(node.getSources()).accept(this, null);
```

Comment:
Do we just want to rename `processWindowFunction` to `processWindowFunctionAndDistinct` so we can call it here since the logic is exactly the same? @shixuan-fan

Reply:
Yeah they are exactly the same but the function name seems too long, yet I cannot come up with a better one :(

Reply:
This duplication is quite trivial. For the readability purposes i would just keep it.